docs/src/examples/ITensor.md

```diff
@@ -17,6 +17,165 @@ or this
 println("T inds = ",inds(T))
 ```
 
+## Getting and Setting Elements of an ITensor
+
+Say we have an ITensor constructed as:
+
+```julia
+i = Index(3,"index_i")
+j = Index(2,"index_j")
+k = Index(4,"index_k")
+
+T = ITensor(i,j,k)
+```
+
+An ITensor constructed this way starts with all of its elements
+equal to zero. (Technically it allocates no storage at all but this is
+an implementation detail.)
+
+**Setting Elements**
+
+To set an element of this ITensor, such as the element where `(i,j,k) = (2,1,3)`,
+you can do the following:
+
+```julia
+T[i=>2,j=>1,k=>3] = -3.2
+```
+
+In the Julia language, the notation `a=>b` is a built-in notation for making a `Pair(a,b)`
+object.
+
+Because the Index objects are passed to `T` along with their values, passing them in a different order has exactly the same effect:
+
+```julia
+# Both of these lines of code do the same thing:
+T[j=>1,i=>2,k=>3] = -3.2
+T[j=>1,k=>3,i=>2] = -3.2
+```
+
+**Getting Elements**
+
+You can retrieve individual elements of an ITensor by accessing them through the same notation used to set elements:
+
+```julia
+el = T[j=>1,i=>2,k=>3]
+println("The (i,j,k) = (2,1,3) element of T is ",el)
+```
+
+## Arithmetic With ITensors
+
+ITensors can be added and subtracted and multiplied by scalars just like plain tensors can. But ITensors have the additional feature that you can add and subtract them even if their indices are in a different order from each other, as long as they have the same collection of indices.
+
+For example, say we have ITensors `A`, `B`, and `C`:
+```julia
+i = Index(3,"i")
+j = Index(2,"j")
+k = Index(4,"k")
+
+A = randomITensor(i,j,k)
+B = randomITensor(i,j,k)
+C = randomITensor(k,i,j)
+```
+Above we have initialized these ITensors to have random elements, just for the sake of this example.
+
+We can then add or subtract these ITensors
+
+```julia
+R1 = A+B
+R2 = A-B
+R3 = A+B-C
+```
+
+or do more complicated operations involving real and complex scalars too:
+
+```julia
+R4 = 2.0*A - B + C/(1+1im)
+```
+
+## Elementwise Operations on ITensors
+
+ITensor objects support Julia broadcasting operations, making it quite easy to carry out element-wise operations on them in a very similar way as for regular Julia arrays. As a concrete example, consider the following ITensor initialized with random elements
+
+```julia
+i = Index(2,"i")
+j = Index(3,"j")
+
+T = randomITensor(i,j)
+```
+
+Here are some examples of basic element-wise operations we can do using Julia's dotted operator broadcasting syntax.
+
+```julia
+# Multiply every element of `T` by 2.0:
+T .*= 2.0
+```
+
+```julia
+# Add 1.5 to every element of T
+T .+= 1.5
+```
+
+The dotted notation works for functions too:
+
+```julia
+# Replace every element in T by its absolute value:
+T .= abs.(T)
+```
+
+```julia
+# Replace every element in T by the number 1.0
+T .= one.(T)
```

Comment:
This operation also is a bit strange for QN ITensors:
```julia
i = Index([QN(0) => 1, QN(1) => 1])
A = emptyITensor(i', dag(i))
A[1, 1] = 2
A .= one.(A)
@show A
```
gives:
```julia
A = ITensor ord=2
Dim 1: (dim=2|id=210)' <Out>
 1: QN(0) => 1
 2: QN(1) => 1
Dim 2: (dim=2|id=210) <In>
 1: QN(0) => 1
 2: QN(1) => 1
NDTensors.BlockSparse{Float64, Vector{Float64}, 2}
 2×2
Block(1, 1)
 [1:1, 1:1]
 1.0
```
Again, the question would be if this should fill in the other possible QN value. Personally I think it should just be ill-defined for QN ITensors. I think the "rule" would be that for `A .= f.(B)`, for QN ITensors, if `f(0) != 0` then the operation is ill-defined, and we should provide a different function like `nonzeros(A) .= f.(nonzeros(B))` if you want the operation to only act on the nonzero values. This logic is a bit tricky to implement the way the ITensor broadcasting is written right now, it needs a rewriting.

Anyway I know this section is about dense tensors where these operations should be fine, I can raise a separate issue but maybe it is just worth making a warning about the QN version of this here.

Reply:
Yes as a concrete step I'll update that docs section to note that it applies to dense tensors only. Good catch!

See my comment above for another idea of how we could define broadcasting of functions on QN ITensors. There are multiple ways we could define it, so we should discuss more.